### ISSUE TITLE
Get rid of NodeTypeConfigurationEnrichmentAspect

### ISSUE THREAD

**Issue by bwaidelich** [neos/neos-development-collection#2867]
The [NodeTypeConfigurationEnrichmentAspect](https://github.com/neos/neos-development-collection/blob/master/Neos.Neos/Classes/Aspects/NodeTypeConfigurationEnrichmentAspect.php) hooks into the `NodeType::__construct` and post-processes the node type configuration in three steps:
* `addEditorDefaultsToNodeTypeConfiguration`
* `addLabelsToNodeTypeConfiguration`
* `mapIconNames`

Using AOP for this seems needlessly complex and hard to debug. Furthermore it's no longer possible to adjust this behavior.

I would suggest to create three [NodeTypePostprocessors](https://github.com/neos/neos-development-collection/blob/master/Neos.ContentRepository/Classes/NodeTypePostprocessor/NodeTypePostprocessorInterface.php) as a replacement for the AOP aspect.

**Comment by bwaidelich** [comment]
It was relatively easy to cover Editor Defaults and Icon names.
But we do some crazy stuff with the i18n there because the string `i18n` is "context sensitive", i.e. can't be converted from the merged node type configuration.

IMO a clean solution to that would be to replace the "i18n" magic with prefixed values like `i18n:Neos.Neos:Inspector:groups.visibility` in the Node Type configurations.

But this would quite a breaking change of course..

To make the current magic work with post processors we'd probably have to go through the super types for each node..

**Comment by skurfuerst** [comment]
Will be obsoleted by ES CR / Neos 9

**Comment by mhsdesign** [comment]
yes it is true, that we dont use an aspect anymore but its still odd to me that its not implemented as postprocessor

https://github.com/neos/neos-development-collection/blob/3fb0d57fd568539f746ee642b944319f8ba8f562/Neos.ContentRepositoryRegistry/Classes/Factory/NodeTypeManager/DefaultNodeTypeManagerFactory.php#L25

**Comment by skurfuerst** [comment]
Hey,

Yeah there was a reason for that - I would love to get rid of it as well. If I remember right it was that the i18n label adjustments needed to happen BEFORE super type inheritance is happening; and the node type post processors did not allow this (as they ran after merging etc).

So if you come up with another solution that would be really nice :) 

All the best,
Sebastian
